public/pages/CreateTrigger/components/CompositeTriggerCondition/ExpressionEditor.js

```diff
@@ -0,0 +1,68 @@
+import React, { useEffect, useState } from 'react';
+import * as _ from 'lodash';
+import { FormikCodeEditor } from '../../../../components/FormControls';
+
+const ExpressionEditor = ({
+  values,
+  formikFieldName,
+  formikFieldPath,
+  isDarkMode = false,
+  triggerIndex,
+}) => {
+  const [editorValue, setEditorValue] = useState('');
+  const formikFullFieldName = `${formikFieldPath}${formikFieldName}`;
+  const formikFullCodeFieldName = _.replace(
+    `${formikFullFieldName}_${triggerIndex}_code`,
+    /[.\[\]]/gm,
+    '_'
+  );
+
+  useEffect(() => {
+    const code = _.get(values, formikFullFieldName, '');
+    _.set(values, formikFullCodeFieldName, code);
+    setEditorValue(code);
+  }, [values]);
+
+  const isInvalid = (name, form) => !form.values[name]?.length;
+
+  const hasError = (name, form) => {
+    return !form.values[name]?.length && 'Invalid condition.';
+  };
+
+  const validate = (value) => {
+    if (!value?.length) return 'Invalid condition.';
+  };
```

Comment:
Can you use these here?
https://github.com/opensearch-project/alerting-dashboards-plugin/blob/main/public/utils/validate.js#L13-L18

Reply:
Those utilities check for error, whereas these check for value